### PR TITLE
bext: make 'single click to definition' an advanced setting

### PR DESCRIPTION
**Pull request by taras-yemets**
Closes [#30437](https://github.com/sourcegraph/sourcegraph/issues/30437)

Makes click to the definition functionality opt-in from advanced settings.

By default, the **Single click to go to definition** checkbox in browser extension advanced settings is not checked. If the user wants to go to the definition, they should click the **Go to definition** button in the IntelliSense popup. 
If the user checks this checkbox, clicking on a code token will take them to the corresponding definition. 

https://user-images.githubusercontent.com/25318659/152305056-29cdc7d0-084d-488d-826f-9847f4b9556e.mp4

### CODE REVIEW

**Comment by jjinnii** [comment]
Amazing work @taras-yemets ! 🎉  Could we change the copy to "Enable click to go to definition"? Otherwise looks great!

**Review by jjinnii — APPROVED** [review]

**Comment by jjinnii** [comment]
@taras-yemets When will this feature be available? I need to communicate back to the customer rep, and I know bext follows a different release cycle.

**Comment by taras-yemets** [comment]
> @taras-yemets When will this feature be available? I need to communicate back to the customer rep, and I know bext follows a different release cycle.

The feature is developed and manually tested.
@erzhtor and I were thinking about adding a few integration tests to this PR, then merge it and plan a new release. I expect this to happen next week.
But still, the integration tests setup is not clear to me, so it may take some time to finalize.

@jjinnii how urgent this release is? If it is quite urgent, we can ship what we have now and create a follow-up issue on tests.

**Comment by jjinnii** [comment]
@taras-yemets Not urgent! I just wanted to get a rough date so they can communicate to the customer. :)

**Comment by taras-yemets** [comment]
@jjinnii I think it could be the end of next week.
WDYT @erzhtor?

**Review by erzhtor — COMMENTED** [review]
Great job! Left a few comments, and also:
- Probably we should remove https://github.com/sourcegraph/sourcegraph/blob/27fc70651629d10743911a266b75c259142a9bd2/client/browser/src/integration/__fixtures__/git_hub_shows_hover_tooltips_when_hovering_a_token/shows_hover_tooltips_when_hovering_a_token_1368546183/recording.har since it is not needed anymore

**Review by erzhtor — APPROVED** [review]
Tremendous job! 💪

File: client/browser/src/integration/github.test.ts [hunk 1]
```diff
@@ -77,6 +84,13 @@ describe('GitHub', () => {
                     },
                 },
             }),
+            SiteProductVersion: () => ({
```
[the remark this answers is not in the document]
Comment (taras-yemets) [reply client/browser/src/integration/github.test.ts]: Removed this mock as redundant.

**Comment by taras-yemets** [comment]
> Tremendous job! muscle

Thanks a lot for your support @erzhtor!

**Pull request merged**